### DISCUSSION TITLE
how init DB_POOL with Actix

### DISCUSSION THREAD

**Discussion by panviktor** [SeaQL/sea-orm#1383] [Q&A] [answered]
Hello, tell me how to be in such a scenario, I want to have access to DbConn in different files using Actix. For correct work I have to use

let conn = web::Data::new(db);

and db must be unreferenced, which prevents me from using

```
lazy_static! {
     static ref DB_POOL: AsyncOnce<DbConn> = AsyncOnce::new(async {
         establish_connection().await.unwrap()
     });
}
```

in diesel this is done via crate r2d2

I don't want to transfer the connection every time and I don't want to pass everything through actix

how can i do it right?

**Comment by billy1624** [comment, marked as answer]
Hey @panviktor, welcome to SeaQL. This is possible, I just tried:

```rs
lazy_static::lazy_static! {
    static ref DB: DatabaseConnection = {
        async_std::task::block_on(async {
            let db_url = env::var("DATABASE_URL").expect("DATABASE_URL is not set in .env file");
            Database::connect(&db_url).await.unwrap()
        })
    };
}
```

**Reply by panviktor** [reply]
it doesn't work for me, I still get the same type - ref DB

I'm just learning how to grow in rust and maybe I don't understand something

**Reply by billy1624** [reply]
Hey @panviktor, what do you mean? I'm confused.

`DatabaseConnection` itself is a pooled connection.

**Reply by panviktor** [reply]
> Hey @panviktor, what do you mean? I'm confused.
> 
> `DatabaseConnection` itself is a pooled connection.

I didn't understand how, but this morning your code worked! Thanks a lot! You helped me a lot!

**Reply by billy1624** [reply]
Any time :)
